package.json

```
@@ -1292,6 +1308,16 @@
         "key": "ctrl+shift+v",
         "mac": "cmd+shift+v",
         "when": "javaLSReady && editorLangId == java"
+      },
+      {
+        "command": "java.action.smartSemicolonDetection",
+        "key": ";",
+        "when": "editorFocus && javaLSReady && editorLangId == java"
```

Try to insert `;` in a `.class` file. The smart semicolon detection runs because the `java` language id includes files ending with `.class`. You'll see this stacktrace.

```
[Error - 16:05:26] Jul. 18, 2023, 4:05:26 p.m. Cannot invoke "org.eclipse.jdt.core.ICompilationUnit.getBuffer()" because "unit" is null
Cannot invoke "org.eclipse.jdt.core.ICompilationUnit.getBuffer()" because "unit" is null
java.lang.NullPointerException: Cannot invoke "org.eclipse.jdt.core.ICompilationUnit.getBuffer()" because "unit" is null
	at org.eclipse.jdt.ls.core.internal.handlers.SmartDetectionHandler.getLocation(SmartDetectionHandler.java:65)
	at org.eclipse.jdt.ls.core.internal.JDTDelegateCommandHandler.executeCommand(JDTDelegateCommandHandler.java:180)
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceExecuteCommandHandler$1.run(WorkspaceExecuteCommandHandler.java:230)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceExecuteCommandHandler.executeCommand(WorkspaceExecuteCommandHandler.java:220)
	at org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.lambda$4(JDTLanguageServer.java:597)
	at org.eclipse.jdt.ls.core.internal.BaseJDTLanguageServer.lambda$0(BaseJDTLanguageServer.java:87)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
```

Ideally we should find something for the `when` condition that includes only `.java` files.

---

I have fixed this issue in the code. See https://github.com/redhat-developer/vscode-java/pull/3159/commits/70ed480dd30b432e6caebb9602fc8c786be0b043#diff-bb57a3bd912abc3ec2e729cb8a743838487677a5517683d0c8913a3619ac296aR368

---

Ok. This is fine. Only slight difference is the semicolon no longer triggers the "Cannot edit in read-only editor" tooltip since we're just silently ignoring it but it's minor.